### ISSUE TITLE
terminologies.txt への追加：訳語に迷う用語について

### ISSUE THREAD

**Issue by masa-ita** [tfug/proofreading#46]
site/ja/tutorials/text/transformer.ipynb を訳しているのですが、訳語に迷うものがいくつもあります。
ディスカッションをお願いします。とりあえず私が訳してみた訳語は次の通りです。カタカナ語がほとんどですが、棒引きや中黒の有無を含めて統一できないかと思っております。
-----
transformer -> Transformer
attention -> アテンション
self-attention -> セルフ・アテンション
sequence -> シーケンス
time step -> タイムステップ
subword -> サブワード
tokenizer -> トークナイザ
graph mode -> グラフモード
graph tensor -> グラフテンソル
unicode -> Unicode
eager tensor -> Eager テンソル
positional encoding -> 位置エンコーディング
attention logit -> アテンション・ロジット
look-ahead mask -> ルック・アヘッド・マスク
scaled dot-product attention -> スケール済み内積アテンション
scaling -> スケーリング
softmax -> ソフトマックス
attention weight -> アテンションの重み
multi-head attention -> マルチ・ヘッド・アテンション
attention head -> アテンション・ヘッド
point-wise -> ポイント・ワイズ
feed forward network -> フィード・フォワード・ネットワーク
encoder -> エンコーダー
decoder -> デコーダー
sub layer -> サブ・レイヤー
layer normalization -> レイヤー正規化
padding mask -> パディング・マスク
state of the art -> ステート・オブ・ジ・アート
teacher forcing -> ティーチャー・フォーシング
beam search -> ビーム・サーチ

**Comment by sfujiwara** [comment]
Attention 周りは「注意機構」とか「自己注意」みたいな訳をちょくちょくみるけど、どれくらい一般的なのかしら

**Comment by masa-ita** [comment]
「注意機構」「自己注意（機構）」は、人工知能学会監修の「深層学習」https://www.amazon.co.jp/dp/476490487X/ でも使われていたと思います。
ちなみに私の初稿では初出の際のみ アテンション（注意機構）、セルフ・アテンション（自己注意）などとしています。

**Comment by AseiSugiyama** [comment]
attention, self attention についてその訳し方すごく良いと思います！

次の単語について、・は次のように一部削除できそうに思います。

look-ahead mask -> ルックアヘッド・マスク
multi-head attention -> マルチヘッド・アテンション
point-wise -> ポイントワイズ
beam search -> ビームサーチ

**Comment by sfujiwara** [comment]
何となく僕もビームサーチとか諸々は「・」無しかなーというので同意なんですが、根拠は？と聞かれると説明できないので何でだろうという気持ちになってます

**Comment by masa-ita** [comment]
ルールが有るかというと難しいですね。原文がダッシュで繋がれているものは中黒（・）なしで良い気がしますが、そうするとこんなかんじになります。

self-attention -> セルフアテンション
time-step -> タイムステップ
multi-head attention -> マルチヘッド・アテンション
look-ahead mask -> ルックアヘッド・マスク
point-wise -> ポイントワイズ
teacher-forcing -> ティーチャーフォーシング

一方で、
beam search -> ビーム・サーチ

**Comment by AseiSugiyama** [comment]
カタカナ複合語のルールがないか調べてみました。[日本翻訳連盟のスタイルガイド](https://www.jtf.jp/jp/style_guide/styleguide_top.html) では次のいずれかを推奨しています。

1. ・(中黒)で区切る
2. 半角スペースで区切る

また、実際の翻訳においては、次の内容を参考にして表記基準を翻訳者が定めることを想定しています。

```text
・表記基準を定めたら、一貫してその基準を運用する。
・用例集、用語集を活用して、表記の統一を図る。
・カタカナ語を安易につないで長くすることはできるだけ避ける。
・語の区切りを入れる方が望ましいが、区切りを多用しない。原語がハイフンでつながれている場合、それ
ぞれの語の独立性が希薄で判読に困難がない場合、語句が短い場合には区切りを入れない方が望ましい。
・区切りを入れないカタカナ複合語の例
ログイン、チェックアウト、リアルタイム、ダブルクリック、スタイルガイド、スピードアップ、バーコード
```

これに照らし合わせると、beam search は語句が短い場合に相当するので、中黒なしで良さそうです。また、独自基準として今回の翻訳で作成したものを運用すれば十分かと思います。

**Comment by masa-ita** [comment]
@AseiSugiyama さん、素晴らしい資料のご紹介ありがとうございます。
ご紹介していただいたスタイルガイドには「カタカナ語末尾の長音は省略しない」というものもありましたね。これらを踏まえて修正してみると、次のようになります。

transformer -> Transformer
attention -> アテンション
self-attention -> セルフアテンション
sequence -> シーケンス
time step -> タイムステップ
subword -> サブワード
tokenizer -> トークナイザー
graph mode -> グラフモード
graph tensor -> グラフテンソル
unicode -> Unicode
eager tensor -> Eager テンソル
positional encoding -> 位置エンコーディング
attention logit -> アテンション・ロジット
look-ahead mask -> ルックアヘッド・マスク
scaled dot-product attention -> スケール済み内積アテンション
scaling -> スケーリング
softmax -> ソフトマックス
attention weight -> アテンションの重み
multi-head attention -> マルチヘッド・アテンション
attention head -> アテンション・ヘッド
point-wise -> ポイントワイズ
feed forward network -> フィードフォワード・ネットワーク
encoder -> エンコーダー
decoder -> デコーダー
sub layer -> サブレイヤー
layer normalization -> レイヤー正規化
padding mask -> パディング・マスク
state of the art -> ステート・オブ・ジ・アート
teacher-forcing -> ティーチャーフォーシング
beam search -> ビームサーチ

個人的には「ステート・オブ・ジ・アート」がちょっと気に入らないのですが、この分野ではSoTAという略語も使われていますし、初出時に「SoTA（訳注：State of The Art、その時点での最高性能）」とでもしたらどうかと思いました。
あと、ティーチャーフォーシングについても「ティーチャーフォーシング（訳注：シーケンスを予測するモデルの訓練時に、過去の予測値ではなく正解値を使うこと）」のようにしたらどうでしょう。

**Comment by masa-ita** [comment]
いろいろ意見を頂戴しましたが、terminologies.txtへの追加のPRを出す前に、transformer.ipynbのPRを出して、文脈中でレビューをしていただきたいと考えています。
なお、上記のティーチャーフォーシングについては、transformer.ipynbの本文に説明が入っているので、訳注を付ける必要がないと判断しました。
